doc/developer-notes.md

```diff
@@ -36,6 +40,63 @@ class Class
 }
 ```
 
+- Strict formatting is enforced all files listed in "src/.formatted-files".
+  Your pull request will fail the Travis build if one of these files is not formatted properly
+
+- To auto-format a file:
+
+   Formatting happens in two passes.  First a python script is run that finds trailing comments that exceed 120 characters
+   and moves them up to the prior line.  Second clang-format-3.8 is executed.
+
+   The first step is to install clang-format.  You likely need to install this exact version or formatting may be slightly different.
+```bash
+sudo apt-get install clang-format-3.8
+```
+
+   Next, run the wrapper script to format your files:
+
+```bash
+./contrib/devtools/clang-format.py format clang-format-3.8 <filenames...>
+```
+   Do not fix other people's formatting issues in your PR, since it distracts from the content!  Let the BitcoinUnlimitedJanitor clean up other people's mess.
+
+
+- To check proper formatting of a file:
+```bash
+./contrib/devtools/clang-format.py check clang-format-3.8 <filenames...>
+```
+
+- To check that your formatting will pass Travis:
+```bash
+make check-formatting
+```
+
+- Formatting style for emacs (add this to your .elisp file).  You can also find a full-featured emacs environment [here](https://github.com/gandrewstone/BUtools).
```

Comment:
I would add also this 
```
(load "<path-to-clang>/tools/clang-format/clang-format.el")
(global-set-key [C-M-tab] 'clang-format-region)
```

this will let you format an entire file or even just the selected area.